uxarray/grid/coordinates.py

```diff
@@ -296,6 +150,266 @@ def _construct_face_centroids(node_x, node_y, node_z, face_nodes, n_nodes_per_fa
     return _normalize_xyz(centroid_x, centroid_y, centroid_z)
 
 
+def _populate_face_centerpoints(grid, repopulate=False):
```

Comment:
I think we used the term `cache` instead of `repopulate` for similar cases ; should we use the same here? @philipc2 can have a better view on this

Reply:
`repopulate` is the correct term in the context of this function (as we've done with the other internal construction methods), since these check if there already exist some coordinate, with `repopulate=True` indicated to ignore any cached values.

This method will always cache the computed result.